assembly/runtime.ts

```diff
@@ -262,15 +259,14 @@ function serializeHeaders(headers: Headers): ArrayBuffer {
     let header = headers[i];
     // i'm sure there's a better way to copy, i just don't know what it is :/
     let wrappedKey = Uint8Array.wrap(header.key);
-    let keyData = data.subarray(currentOffset, wrappedKey.byteLength);
+    let keyData = data.subarray(currentOffset, currentOffset + wrappedKey.byteLength);
```

Comment:
The fix here is just because the subarray function takes begin and end as the parameters and not the begin and size.